tests/framework/extensions/kubeapi/secrets/create.go

```diff
@@ -8,18 +8,22 @@ import (
 	"github.com/rancher/rancher/tests/framework/extensions/unstructured"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
 )
 
-// CreateSecret is a helper function that uses the dynamic client to create a secret on a namespace for a specific cluster.
-// It registers a delete function.
-func CreateSecret(client *rancher.Client, secret *corev1.Secret, clusterID, namespace string) (*corev1.Secret, error) {
+// CreateSecretForCluster is a helper function that uses the rancher client to create a secret in a namespace for a specific cluster.
+func CreateSecretForCluster(client *rancher.Client, clusterID, namespace string, secret *corev1.Secret) (*corev1.Secret, error) {
```

Comment:
What's the difference between this func, and other `CreateSecret` func?

Reply:
The difference being that this function just wraps the functionality for creating the dynamic client for the cluster, so that in my encryption key rotation tests I don't have to recreate the client in the creation loop. I don't know if it creates a measurable performance impact, but it takes like an hour to generate 10k secrets so figured it was probably better to do it this way.